scipy/stats/kde.py

```diff
@@ -102,17 +106,24 @@ class gaussian_kde(object):
 
     Scott's Rule [1]_, implemented as `scotts_factor`, is::
 
-        n**(-1./(d+4)),
+        neff**(-1./(d+4)),
 
-    with ``n`` the number of data points and ``d`` the number of dimensions.
-    Silverman's Rule [2]_, implemented as `silverman_factor`, is::
+    with ``neff`` the effective number of data points and ``d`` the number of
+    dimensions.  Silverman's Rule [2]_, implemented as `silverman_factor`, is::
 
-        (n * (d + 2) / 4.)**(-1. / (d + 4)).
+        (neff * (d + 2) / 4.)**(-1. / (d + 4)).
```

Comment:
You can't just change this, for backwards compatibility reasons.

Reply:
also unclear why `neff` is introduced as a parameter if you can just calculate it from `weights`.

Reply:
I spent a bit of time thinking about this. It doesn't break backward compatibility functionally, since `neff=n` in the default equally weighted case. However I agree that this is confusing in terms of the documentation, which I have updated back to the original, with an extra statement for the unequally weighted case.

Reply:
I put `neff` as an attribute, as it's generally a useful number to access as a user. For example, it can be fed into additional statistical analyses in place of `n` for the unequally weighted case.

Reply:
If Neff is a computed attribute, maybe make it a read-only property?

Reply:
I'm happy to implement this, but it's not very consistent with the rest of the class. By that logic, pretty much all of the attributes should be read-only, as they are all computed from `dataset`. Perhaps it would be good to make all of the attributes read-only as well?

Reply:
I'd vote for not bothering with read-only.

Reply:
Agreed with not bothering with read-only